ampd/src/starknet/json_rpc.rs

```diff
@@ -107,32 +110,35 @@ where
     }
 
     // Fetches a transaction receipt by hash and extracts a `SignersRotatedEvent` if present
-    async fn get_event_by_hash_signers_rotated(
+    async fn get_event_by_hash_and_index_signers_rotated(
         &self,
-        tx_hash: CheckedFelt,
+        message_id: FieldElementAndEventIndex,
     ) -> Result<Option<(Felt, SignersRotatedEvent)>> {
```

Comment:
It seems you don't use the `Felt` in the tuple.
Does it make sense to mimic the contract call get function and return only `Option<ContractCallEvent>`?

This will reduce the code, remove errors entirely and make the code more consistent.